File: Task_1/Task/Square_Root/Square_Root.c
#include <stdio.h>
double squareRoot(double num, int iterations);

int main() {
    printf("Hello\n");
    printf("%f\n", squareRoot(100, 10));  
    return 0;
}


double squareRoot(double num, int iterations) {
    // if(num <= 0){return;}
    double currentGuess = num/2;
    for (int i = 0; i < iterations; i++){
        double newGuess = (currentGuess + num/currentGuess) * 0.5;
        currentGuess = newGuess;
    }
    return currentGuess;
    
}

File: Task_1/Square_Root/Square_Root.py
def squareRoot(num,iterations = 10):
    if num < 0:
        return None
    currentGuess = num / 2
    for i in range(iterations):
        newGuess = 0.5 * (currentGuess + num/currentGuess)
        currentGuess = newGuess
    return currentGuess